src/algokit/core/sandbox.py

```diff
@@ -192,6 +195,18 @@ def check_docker_compose_for_new_image_versions(self) -> None:
                 "algod has a new version available, run `algokit localnet reset --update` to get the latest version"
             )
 
+    def save_current_directory(self) -> None:
+        current_dir_file = get_app_config_dir() / "localnet_directory.txt"
```

Comment:
Potentially we could use docker itself to track this state via `docker compose ls --all --filter name='algokit_sandbox*' --format json`. It would be fairly easy for this state file to get out of sync if the user interacts directly with docker.